### PR TITLE
TruLens as namespace package 

### PR DESCRIPTION
**Pull request by sfc-gh-chu**
# Description

Major refactor of the trulens_eval package. This introduces some breaking changes
 - Factors out the following packages into independent packages that share the trulens namespace
   - `trulens.core`: essential recording functionality, base classes for app types, feedback, and provider base classes
   - `trulens.feedback`: library for feedback evaluation including the bulk of what was previously `trulens.feedback`. 
   - `trulens.dashboard`: code around defining and managing dashboard. Includes react components and build.
   - `trulens.instrument.llamaindex`: code for TruLlama and instrumentation
   - `trulens.instrument.langchain`: code for TruChain and instrumentation
   - `trulens.instrument.nemo`: code for TruRails
   - `trulens.benchmark`: code for feedback benchmarking and metaevaluation. Does not include datasets. 
   - `trulens.providers.*`: code for optional providers like OpenAI, Cortex, HuggingFace

 - Rename trulens_eval -> trulens
 - Uses poetry and setuptools for managing the building and package metadata.
 - Updated docs with autogenerated API reference, dark mode, examples tab
 - Updated build and release pipeline

## Other details good to know for developers
 - [x] Need to change the az pipeline
 - [ ] Documentation and examples should have been updated with the new import scheme, but may have missed some.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [x] This change requires a documentation update

### CODE REVIEW

**Comment by sfc-gh-jreini** [comment]
For trulens.feedback, it might be better to split into different provider packages, such as:

- trulens.feedback.openai
- trulens.feedback.huggingface
- trulens.feedback.litellm
- trulens.feedback.bedrock

**Comment by sfc-gh-jreini** [comment]
Can we push off `trulens_explain` to its own package here and move `trulens_eval/src` to just `src`?

**Comment by sfc-gh-jreini** [comment]
What are the updates to the deployment process with this change to poetry?

File: README.md [hunk 1]
```diff
@@ -1,33 +1,12 @@
...
+trulens_eval/README.md . If you are editing README.md, your changes will be overwritten.
 -->
-
-![PyPI - Version](https://img.shields.io/pypi/v/trulens_eval?label=trulens_eval&link=https%3A%2F%2Fpypi.org%2Fproject%2Ftrulens-eval%2F)
-![Azure DevOps builds (job)](https://img.shields.io/azure-devops/build/truera/5a27f3d2-132d-40fc-9b0c-81abd1182f41/9)
-![GitHub](https://img.shields.io/github/license/truera/trulens)
-![PyPI - Downloads](https://img.shields.io/pypi/dm/trulens_eval)
-[![Slack](https://img.shields.io/badge/slack-join-green?logo=slack)](https://communityinviter.com/apps/aiqualityforum/josh)
-[![Docs](https://img.shields.io/badge/docs-trulens.org-blue)](https://www.trulens.org/trulens_eval/getting_started/)
-[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/truera/trulens/blob/releases/rc-trulens-eval-0.17.0/trulens_eval/examples/quickstart/colab/langchain_quickstart_colab.ipynb)
```
Comment (sfc-gh-jreini) [inline README.md:13]: Why delete the shield icons?

File: README.md [hunk 2]
```diff
@@ -1,33 +1,12 @@
...
-
-TruLens provides a set of tools for developing and monitoring neural nets,
-including large language models. This includes both tools for evaluation of LLMs
-and LLM-based applications with _TruLens-Eval_ and deep learning explainability
-with _TruLens-Explain_. _TruLens-Eval_ and _TruLens-Explain_ are housed in
-separate packages and can be used independently.
-
-The best way to support TruLens is to give us a ⭐ on
-[GitHub](https://www.github.com/truera/trulens) and join our [slack
-community](https://communityinviter.com/apps/aiqualityforum/josh)!
```
Comment (sfc-gh-jreini) [inline README.md:25]: Keep this, just move it down below the header

File: examples/expositional/end2end_apps/trubot/trubot.py [hunk 3]
```diff
@@ -3,31 +3,30 @@
...
 import numpy as np
 import openai
 import pinecone
 from slack_bolt import App
 from slack_sdk import WebClient
+from trulens.core import Feedback
+from trulens.core import Select
+from trulens.core import Tru
+from trulens.core.schema.feedback import FeedbackMode
+from trulens.langchain import TruChain
```
Comment (sfc-gh-jreini) [inline examples/expositional/end2end_apps/trubot/trubot.py:17]: Let's put lanchain, llama-index-etc in a category. Something like trulens.frameworks.langchain, trulens.frameworks.llama_index.

This will more clearly distinguish from langchain feedback provider, etc.

File: examples/expositional/frameworks/llama_index/llama_index_groundtruth.ipynb [hunk 4]
```diff
@@ -41,8 +41,9 @@
     "from llama_index.readers.web import SimpleWebPageReader\n",
     "import openai\n",
     "\n",
-    "from trulens_eval import TruLlama, Feedback, Tru, feedback\n",
-    "from trulens_eval.feedback import GroundTruthAgreement\n",
+    "from trulens.core import Tru, Feedback\n",
+    "from trulens.llamaindex import TruLlama\n",
+    "from trulens.external import OpenAI, GroundTruthAgreement\n",
```
Comment (sfc-gh-jreini) [inline examples/expositional/frameworks/llama_index/llama_index_groundtruth.ipynb:46]: Suggesting to do a scheme like: trulens.feedback.openai
trulens.feedback.litellm
...

File: README.md (hunk 1 above)
Comment (sfc-gh-chu) [reply README.md:13]: Oh might be from overriding the root readme with the trulens_eval one. Can keep those

**Comment by sfc-gh-jreini** [comment]
Can we change `ext/` to `community/`?

**Comment by sfc-gh-jreini** [comment]
Love having the examples loaded automatically in docs. Couple nits on this:

- Collapse the experimental examples and move them to the bottom; I'd be okay with removing them if not possible
- Remove the 'Feedback' example (at the top) ([PR to do this](https://github.com/truera/trulens/pull/1319))
- Move Quickstarts to the top of the nav, followed by Expositional

File: Makefile [hunk 5]
```diff
@@ -2,58 +2,130 @@
...
+	poetry run ruff check --fix
+
+# Run the ruff formatter.
+format: .env/create
+	poetry run ruff format
+
+precommit-hooks: .env/create
+	poetry run pre-commit install
+
+typestubs: .env/create $(POETRY_DIRS)
```
Comment (sfc-gh-pmardziel) [inline Makefile:29]: What is this being used for?

**Comment by sfc-gh-chu** [comment]
> * Move Quickstarts to the top of the nav, followed by Expositional

Noticed we symlink the quickstarts on the Getting Started page. Might be confusing to have this in two spots?

File: .azure_pipelines/ci-docs-pr.yaml [hunk 6]
```diff
@@ -23,7 +23,7 @@ jobs:
     strategy:
       matrix:
         latest:
-          python-version: "3.12"
+          python-version: "3.11"
```
Comment (sfc-gh-pdharmana) [inline .azure_pipelines/ci-docs-pr.yaml:26]: Please add in the description why this is needed.

File: src/core/trulens/core/app/base.py [hunk 7]
```diff
@@ -91,8 +93,22 @@
 ```
 """
 
+_component_impls: Dict[str, Type[ComponentView]] = {}
+
+
+class ComponentViewMeta(ABCMeta):
```
Comment (sfc-gh-chu) [inline src/core/trulens/core/app/base.py:99]: Logic change: using metaclasses to add anything that inherits from `ComponentView` to `_component_impls`. 

This replaces definitions from instrumentation extensions here (e.g. LangchainComponentView) here, which would have dependencies. Also should make it easier to add new instrumentation implementations

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 8]
```diff
@@ -4,7 +4,7 @@ about: Thanks for taking the time to fill out this bug report! Please complete t
   following form to help us assist you.
 title: "[BUG]"
 labels: bug
-assignees: yuvneshtruera
+assignees: sfc-gh-jreini
```
Comment (sfc-gh-pdharmana) [inline .github/ISSUE_TEMPLATE/bug-report.md:7]: Thank you, but this should be part of another PR?

**Review by sfc-gh-pdharmana — COMMENTED** [review]
Will continue the review.

File: src/core/trulens/core/app/base.py [hunk 9]
```diff
@@ -676,32 +652,9 @@ def select_context(cls, app: Optional[Any] = None) -> Lens:
         access the retrieved contexts that would appear in a record were these
         components to execute.
         """
```
Comment (sfc-gh-chu) [inline src/core/trulens/core/app/base.py:654]: Logic Change: keeping `App.select_context` abstract for subclasses to implement.

File: src/core/trulens/core/feedback/feedback.py [hunk 10]
```diff
@@ -111,103 +114,17 @@ def __repr__(self):
         return f"InvalidSelector({self.selector})"
```
Comment (sfc-gh-chu) [inline src/core/trulens/core/feedback/feedback.py:115]: this got moved to trulens-feedback

File: src/core/trulens/core/schema/select.py [hunk 11]
```diff
@@ -0,0 +1,146 @@
+"""Serializable selector-related classes."""
```
Comment (sfc-gh-chu) [inline src/core/trulens/core/schema/select.py:1]: Moved this out in an effort to get as many top-level imports as possible and this would cause a circular import in `feedback.py`.

File: src/core/trulens/core/utils/imports.py [hunk 12]
```diff
@@ -151,73 +135,18 @@ def get_package_version(name: str) -> Optional[version.Version]:
...
+beyond this package, you may need to reinstall trulens so that all of the
 dependencies get installed and hopefully corrected:
 
     ```bash
-    pip uninstall -y trulens_eval
-    pip install trulens_eval
+    pip uninstall -y trulens
+    pip install trulens
     ```
 """
```
Comment (sfc-gh-chu) [inline src/core/trulens/core/utils/imports.py:148]: removed check_imports from `__init__` as most hard dependencies should be resolved at install-time so this no longer needed.

File: src/core/trulens/core/tru.py [hunk 13]
```diff
@@ -296,103 +286,10 @@ def _create_snowflake_schema_if_not_exists(
             snowflake_connection_parameters["database"]
         ].schemas.create(schema, mode=CreateMode.if_not_exists)
```
Comment (sfc-gh-chu) [inline src/core/trulens/core/tru.py:288]: API Change: removing usage of `Tru().Chain`, `Tru().Llama`, etc

File: src/core/trulens/core/tru.py [hunk 14]
```diff
@@ -1119,300 +1016,3 @@ def stop_evaluator(self):
             self._evaluator_stop = None
 
         self._evaluator_proc = None
```
Comment (sfc-gh-chu) [inline src/core/trulens/core/tru.py:1018]: dashboard is packaged away from core so needed to be decoupled from Tru()

File: src/feedback/trulens/feedback/v2/feedback.py [hunk 15]
```diff
@@ -155,13 +204,13 @@ class Groundedness(Semantics, WithPrompt):
     # hugs._summarized_groundedness
     # hugs._doc_groundedness
 
-    system_prompt: ClassVar[PromptTemplate] = PromptTemplate.from_template(
+    system_prompt: ClassVar[str] = cleandoc(
```
Comment (sfc-gh-chu) [inline src/feedback/trulens/feedback/v2/feedback.py:207]: Removing langchain as a dependency for `trulens-feedback` so prompts are back to classic strings.

File: .pre-commit-config.yaml [hunk 16]
```diff
@@ -15,13 +15,58 @@ repos:
...
         types_or: [ python, pyi, jupyter ]
 -   repo: https://github.com/srstevenson/nb-clean
     rev: 3.3.0
     hooks:
       - id: nb-clean
         args: [--remove-empty-cells]
+-   repo: https://github.com/tox-dev/pyproject-fmt
+    rev: "2.0.4"
+    hooks:
+      - id: pyproject-fmt
```
Comment (sfc-gh-dkurokawa) [inline .pre-commit-config.yaml:31]: Will this conflict with the L7?

File: .vscode/settings.json [hunk 17]
```diff
@@ -1,6 +1,16 @@
 {
+	"python.analysis.typeCheckingMode": "basic",
+	"python.testing.pytestArgs": [
+		"tests"
+	],
+	"python.testing.unittestEnabled": false,
+	"python.testing.pytestEnabled": true,
+	"ruff.format.preview": true,
```
Comment (sfc-gh-dkurokawa) [inline .vscode/settings.json:8]: the identation of this file seems kinda inconsistent now.

File: DEVELOPMENT.md [hunk 18]
```diff
@@ -0,0 +1,168 @@
...
+
+## Install the TruLens project
+
+Install `trulens` into your environment, including optional dependencies, by running the following command:
+
+```bash
+poetry install
+```
+
+To install the exact environment specified by `poetry.lock` without user-specified dependencies , use the `--sync` flag. This also installs optional dependencies.
```
Comment (sfc-gh-dkurokawa) [inline DEVELOPMENT.md:45]: Sorry I didn't quite understand what `--sync` does, could you elaborate?

File: Makefile [hunk 19]
```diff
@@ -2,58 +2,153 @@
...
+	poetry run pytest --rootdir=. tests/docs_notebooks/*
+
+# Release Steps:
+## Step: Clean repo:
+clean-dashboard:
+	rm -rf src/dashboard/*.egg-info
+
+clean: clean-dashboard
+	rm -rf dist build
+	git clean -fxd
```
Comment (sfc-gh-dkurokawa) [inline Makefile:115]: I feel like this is kinda dangerous? Maybe have a confirmation?

File: meta.yaml [hunk 20]
```diff
@@ -0,0 +1,8 @@
+# conda metadata build recipe
```
Comment (sfc-gh-dkurokawa) [inline meta.yaml:1]: sorry what is this file?

Comment (sfc-gh-chu) [reply meta.yaml]: It's a [conda build recipe](https://docs.conda.io/projects/conda-build/en/stable/concepts/recipe.html) that we'll use to create a conda package

File: src/core/pyproject.toml [hunk 21]
```diff
@@ -0,0 +1,61 @@
...
+merkle-json = "^1.0"
+sqlalchemy = "^2.0"
+alembic = "^1.11"
+nest-asyncio = "^1.5"
+python-dotenv = "^1.0"
+
+snowflake-core = { version = "^0.9.0", python = "<3.12", optional = true }
+snowflake-connector-python = { version = "^3.11.0", python = "<3.12", optional = true }
+snowflake-snowpark-python = { version = "^1.20.0", python = "<3.12", optional = true }
+snowflake-sqlalchemy = { version = "^1.6", python = "<3.12", optional = true }
```
Comment (sfc-gh-pdharmana) [inline src/core/pyproject.toml:46]: We should repackage them as part of `trulens-snowflake` and not part of `trulens-core`?

Comment (sfc-gh-pdharmana) [reply src/core/pyproject.toml:46]: Seems like you put this as an extra in poetry. I guess that is alright.

File: src/dashboard/trulens/dashboard/pages/Evaluations.py [hunk 22]
```diff
@@ -398,7 +400,7 @@ def highlight(s):
                     else:
                         st.text("No feedback details.")
 
-                if selected_fcol != None:
+                if selected_fcol is not None:
```
Comment (sfc-gh-pdharmana) [inline src/dashboard/trulens/dashboard/pages/Evaluations.py:403]: Is this through linter?

File: src/core/trulens/core/app/custom.py [hunk 23]
```diff
@@ -3,15 +3,15 @@
 
 This wrapper is the most flexible option for instrumenting an application, and can be used to instrument any custom python class.
 
-!!! example
+Example:
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/app/custom.py:6]: I thought doing !!!example, will put this in a formatted box for the docs.

File: src/core/trulens/core/app/custom.py [hunk 24]
```diff
@@ -539,6 +538,4 @@ def method(self_class, cls: type, name: str) -> None:
         TruCustomApp.functions_to_instrument.add(getattr(cls, name))
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/app/custom.py:540]: Why did we have these imports before?

File: src/core/trulens/core/database/legacy/migration.py [hunk 25]
```diff
@@ -1,6 +1,6 @@
 """
 This is pre-sqlalchemy db migration. This file should not need changes. It is
-here for backwards compatibility of oldest trulens-eval versions.
+here for backwards compatibility of oldest TruLens versions.
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/database/legacy/migration.py:3]: Do we have any json records in the database, that includes trulens objects? I would guess not, but if there are we would be backwards incompatible with data as well cc: @sfc-gh-pmardziel

File: src/core/trulens/core/feedback/feedback.py (hunk 10 above)
Comment (sfc-gh-pdharmana) [reply src/core/trulens/core/feedback/feedback.py:115]: Thanks

File: src/core/trulens/core/feedback/feedback.py [hunk 26]
```diff
@@ -381,26 +298,20 @@ def _default_selectors(self):
...
-            )
+            selectors = {par_names[0]: Select.RecordOutput}
+            self._print_guessed_selector(par_names[0], Select.RecordOutput)
 
             # TODO: replace with on_output ?
 
         elif len(par_names) == 2:
             # Two arguments remaining. Assume they are record input and output
             # respectively.
             selectors = {
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/feedback/feedback.py:309]: I love your refractor to remove `mod_*` the code is so much cleaner.

File: src/core/trulens/core/schema/app.py [hunk 27]
```diff
@@ -129,18 +128,18 @@ def __init__(
                     # in local files instead of the DB. Leaving here for now as
                     # serialization of large apps might make this necessary
                     # again.
-                    """
-                    path_json = Path.cwd() / f"{app_id}.json"
-                    path_dill = Path.cwd() / f"{app_id}.dill"
+                    # """
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/schema/app.py:131]: Why do we stll need `""""` ?

File: src/core/trulens/core/schema/select.py (hunk 11 above)
Comment (sfc-gh-pdharmana) [reply src/core/trulens/core/schema/select.py:1]: Thanks for doing this.

File: src/core/trulens/core/utils/pyschema.py [hunk 28]
```diff
@@ -146,14 +139,14 @@ class Module(SerialModel):
     package_name: Optional[str] = None  # some modules are not in a package
     module_name: str
 
-    def of_module(mod: ModuleType, loadable: bool = False) -> Module:
+    def of_module(mod: ModuleType, loadable: bool = False) -> "Module":
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/utils/pyschema.py:142]: Is this a change in logic?

File: src/core/trulens/core/utils/python.py [hunk 29]
```diff
@@ -40,14 +41,14 @@
     Future = futures.Future
     """Alias for [concurrent.futures.Future][].
 
-    In python < 3.9, a subclass of [concurrent.futures.Future][] with
+    In python < 3.9, a sublcass of [concurrent.futures.Future][] with
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/utils/python.py:44]: I am so surprised. How did you find these little things in such a large change.

File: src/core/trulens/core/utils/python.py [hunk 30]
```diff
@@ -342,7 +343,7 @@ def wrapper(*args, **kwargs):
 STACK = "__tru_stack"
 
 
-def caller_frame(offset=0) -> frame:
+def caller_frame(offset=0) -> FrameType:
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/utils/python.py:346]: PLease explain the logic chaange here.

File: src/core/trulens/core/instruments.py [hunk 31]
```diff
@@ -133,7 +137,7 @@ def on_new_record(self, func: Callable):
     # Called during invocation.
     def on_add_record(
         self,
-        ctx: RecordingContext,
+        ctx: "RecordingContext",
```
Comment (sfc-gh-pdharmana) [inline src/core/trulens/core/instruments.py:140]: When do we use a string vs Type.

**Review by sfc-gh-pdharmana — COMMENTED** [review]
Reviewed trulens core move.

File: src/core/pyproject.toml (hunk 21 above)
Comment (sfc-gh-chu) [reply src/core/pyproject.toml:46]: How are we thinking of packaging snowflake integrations? We support snowflakedb connection details in `Tru()` but not all users need to integrate with Snowflake. Perhaps `trulens-snowflake` with a `SnowflakeTru()` may be more appropriate

File: src/dashboard/trulens/dashboard/pages/Evaluations.py (hunk 22 above)
Comment (sfc-gh-chu) [reply src/dashboard/trulens/dashboard/pages/Evaluations.py:403]: yup

File: src/core/trulens/core/app/custom.py (hunk 23 above)
Comment (sfc-gh-chu) [reply src/core/trulens/core/app/custom.py:6]: [Either will way will show admonitions on the docs](https://mkdocstrings.github.io/python/usage/docstrings/google/?h=admonition#docstring). Figured this might be more readable as a docstring.

File: src/core/pyproject.toml (hunk 21 above)
Comment (sfc-gh-jreini) [reply src/core/pyproject.toml:46]: Maybe something like `trulens-logging-snowflake` to disambiguate from `trulens-providers-cortex`

File: src/core/trulens/core/app/custom.py (hunk 24 above)
Comment (sfc-gh-chu) [reply src/core/trulens/core/app/custom.py:540]: cc @sfc-gh-pmardziel

File: src/core/trulens/core/instruments.py (hunk 31 above)
Comment (sfc-gh-chu) [reply src/core/trulens/core/instruments.py:140]: In this case, importing `RecordingContext` would cause a circular import. Python has an option to get around this by importing under an `if typing.TYPE_CHECKING` condition (see line 63). code and imports here are only run for type checking and are ignored at runtime. Types referencing imports under `typing.TYPE_CHECKING` are stringified. Most common IDEs and type checkers are able to match it to the appropriate type.

File: src/core/trulens/core/utils/python.py (hunk 30 above)
Comment (sfc-gh-chu) [reply src/core/trulens/core/utils/python.py:346]: I fixed the return type for this method. `inspect.stack()` is a builtin python method that returns `list[FrameInfo]`. `FrameInfo.frame` would be a FrameType so this method should reflect that

File: src/core/trulens/core/utils/pyschema.py (hunk 28 above)
Comment (sfc-gh-chu) [reply src/core/trulens/core/utils/pyschema.py:142]: No just a type hint. Looking at it now, this is actually not necessary.

File: src/core/trulens/core/utils/python.py (hunk 29 above)
Comment (sfc-gh-chu) [reply src/core/trulens/core/utils/python.py:44]: I just made another PR that contains more here but found most of these with some combination of `codespell` and a VS Code spell checking extension.
There are a few false positives so I didn't add it as a precommit hook.
https://github.com/truera/trulens/pull/1327

File: src/core/trulens/core/feedback/feedback.py (hunk 26 above)
Comment (sfc-gh-chu) [reply src/core/trulens/core/feedback/feedback.py:309]: There's quite a lot more cases across the repo so it would be more time consuming to fix but agree it helps with legibility.

File: .azure_pipelines/templates/run-optional-tests.yaml [hunk 32]
```diff
@@ -0,0 +1,24 @@
+parameters:
+- name: testSuite
+  type: string # must be one of: unit, e2e, notebook
```
Comment (sfc-gh-dkurokawa) [inline .azure_pipelines/templates/run-optional-tests.yaml:3]: I think you can use the `values` thing to say it has to be one of those.

File: .azure_pipelines/templates/run-tests.yaml [hunk 33]
```diff
@@ -0,0 +1,9 @@
+parameters:
+- name: testSuite
+  type: string # must be one of: unit, e2e, notebook
```
Comment (sfc-gh-dkurokawa) [inline .azure_pipelines/templates/run-tests.yaml:3]: same as above

File: Makefile [hunk 34]
```diff
@@ -2,58 +2,153 @@
...
+format: env
+	poetry run ruff format
+
+precommit-hooks:
+	poetry run pre-commit install
+
+run-precommit:
+	poetry run pre-commit run --all-files --show-diff-on-failure
 
 # Start a jupyer lab instance.
```
Comment (sfc-gh-dkurokawa) [inline Makefile:43]: nit: can you fix this typo that has nothing to do with this PR haha

File: Makefile [hunk 35]
```diff
@@ -2,58 +2,153 @@
...
-linkcheck: site
+docs-linkcheck: site
 	lychee "site/**/*.html"
+
+# Start the trubot slack app.
+trubot:
+	poetry run python -u examples/trubot/trubot.py
+
+# Run a test with the optional flag set, meaning @optional_test decorated tests
+# are run.
```
Comment (sfc-gh-dkurokawa) [inline Makefile:78]: is this supposed to be describing `coverage`? It kinda does, but it's mostly about the coverage part isn't it?

File: Makefile [hunk 36]
```diff
@@ -2,58 +2,153 @@
...
+
+coverage:
+	ALLOW_OPTIONAL_ENV_VAR=true pytest --rootdir=. tests/* --cov src --cov-report html
```
Comment (sfc-gh-dkurokawa) [inline Makefile:81]: do you not have to say `poetry run pytest`?

File: Makefile [hunk 37]
```diff
@@ -2,58 +2,153 @@
...
+	if [[ $$REPLY =~ ^[Yy]$$ ]]; then \
+		git clean -fxd; \
+	fi;
+
+## Step: Build wheels
+build-dashboard: env clean-dashboard
+	poetry run python -m build src/dashboard -o $(REPO_ROOT)/dist/trulens-dashboard;
+
+build: $(POETRY_DIRS)
+	for dir in $(POETRY_DIRS); do \
```
Comment (sfc-gh-dkurokawa) [inline Makefile:126]: btw I think if you just left the `.ONESHELL` thing you wouldn't need all these `\`.

File: Makefile [hunk 38]
```diff
@@ -2,58 +2,153 @@
...
+	done
+	make build-dashboard
+
+## Step: Upload wheels to pypi
+# Usage: TOKEN=... make upload-trulnes-instrument-langchain
+upload-%:
+	poetry run twine upload -u __token__ -p $(TOKEN) dist/$*/*
+
+upload-all: build
+	poetry run `twine upload --skip-existing -u __token__ -p $(TOKEN) dist/**/*
```
Comment (sfc-gh-dkurokawa) [inline Makefile:148]: what's with the backtick?

**Review by sfc-gh-dkurokawa — APPROVED** [review]

**Comment by sfc-gh-jreini** [comment]
Seeing some issues using the evaluations page. Have you seen this @sfc-gh-chu ?

Screenshot 2024-08-08 at 5 19 21 PM

**Comment by sfc-gh-chu** [comment]
> Seeing some issues using the evaluations page. Have you seen this @sfc-gh-chu ? Running a local trulens deployment off your branch
> 
> Screenshot 2024-08-08 at 5 19 21 PM

If you're installing in editable mode, it'll need to build the react components. I'll add a makefile recipe for that

**Comment by sfc-gh-dhuang** [comment]
I reviewed and made some changes to adapt to the new packaging structure for `trulens.benchmark`.  I've tested and the new benchmark code + new output score space validation logic works e2e
